xlcalculator/xlfunctions/operator.py

```diff
@@ -44,6 +44,12 @@ def OP_EQ(
         left: func_xltypes.XlAnything,
         right: func_xltypes.XlAnything
 ) -> func_xltypes.XlBoolean:
+    if isinstance(left, (str, func_xltypes.Text)):
+        left = left.upper()
+
+    if isinstance(right, (str, func_xltypes.Text)):
+        right = right.upper()
+
```

Comment:
I really do not like this special handling here. I would rather create a function like `canonical()` that converts any type to its canonical form that then gets compared. And for strings `canonical()` returned all upper cased.

So I just checked and `str` is not a valid "func_types.XlAnything"  anyways. Thus you do not even need the canonical function. Instead you overwrite `__eq__()` and `__ne__()`, `__lt__()`, `__le__()`, `__gt__()`, `__ge__()` in `func_xltypes.Text`.